pkg/graveler/graveler.go

```diff
@@ -1006,11 +1132,85 @@ func (g *KVGraveler) Log(ctx context.Context, repositoryID RepositoryID, commitI
 }
 
 func (g *KVGraveler) ListBranches(ctx context.Context, repositoryID RepositoryID) (BranchIterator, error) {
-	return g.db.ListBranches(ctx, repositoryID)
+	return g.RefManager.ListBranches(ctx, repositoryID)
+}
+
+// dropBranchStaging Deletes all staging data (Staging Token + Sealed Tokens) of the given branch
+func (g *KVGraveler) dropBranchStaging(ctx context.Context, branch *Branch) error {
+	var wg multierror.Group
+	tokens := make([]StagingToken, len(branch.SealedTokens)+1)
+	copy(tokens, branch.SealedTokens)
+	tokens[len(tokens)-1] = branch.StagingToken
+
+	for _, st := range tokens {
+		st := st // Pinning
+		wg.Go(func() error {
+			if err := g.StagingManager.Drop(ctx, st); err != nil && !errors.Is(err, ErrNotFound) {
+				return err
+			}
+			return nil
+		})
+	}
+	return wg.Wait().ErrorOrNil()
```

Comment:
A smoother way of dropping a branch workspace, would be to replace the staging token (or drop just that) and erase all its sealed tokens (while dropping in the background). `Drop` isn't an atomic operation, and even after we succeed we'll end up with empty sealed tokens which is meaningless.

Reply:
I see it's only being used during `DeleteBranch`, since the `Branch` entry is on its way to be removed, my previous comment is of less significance.